### PR TITLE
Fix fetch/list entry functions to only return non-expirable and unexpired entries

### PR DESCRIPTION
**Pull request by ryysud**
Signed-off-by: Ryuma Yoshida <ryuma.y1117@gmail.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

This PR affects the functionality of `FetchRegistrationEntry` and `ListRegistrationEntries` methods in [datastore.proto](https://github.com/spiffe/spire/blob/41eb0c78bdeee83b11e13fdcc8b4a6e55620b318/proto/spire/server/datastore/README_pb.md#datastore) .

**Description of change**

I updated some sqls to only return non-expirable and unexpired registration entries.

The functions to get current unix timestamp in SQLite3, PostgreSQL and MySQL are as below:

- https://www.sqlite.org/lang_datefunc.html
- https://www.postgresql.org/docs/12/functions-datetime.html
- https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html

**Which issue this PR fixes**

This PR fixes https://github.com/spiffe/spire/issues/1071 .

### CODE REVIEW

**Comment by azdagron** [comment]
Hi @ryysud! Thanks for the contribution! I'm getting some feedback and clarification on #1071 that will help me form an opinion on this PR. At a minimum I think we'll want this filtering to be opt-in, since not showing these entries can be confusing in other cases.

Thank you for your patience and thanks again for the contribution!

**Comment by azdagron** [comment]
Clarification on #1071 might also inform on whether or not the datastore is the proper layer at which to implement this functionality.

File: pkg/server/plugin/datastore/sql/sql.go [hunk 1]
```diff
@@ -1150,7 +1150,7 @@ func buildFetchRegistrationEntryQuery(dbType string, req *datastore.FetchRegistr
 func buildFetchRegistrationEntryQuerySQLite3(req *datastore.FetchRegistrationEntryRequest) (string, []interface{}, error) {
 	const query = `
 WITH listing AS (
-	SELECT id FROM registered_entries WHERE entry_id = ?
+	SELECT id FROM registered_entries WHERE entry_id = ? AND (expiry = 0 OR expiry >= strftime('%s','now'))
```
Comment (amoore877) [inline pkg/server/plugin/datastore/sql/sql.go]: a Server User may be checking if some registration(s) exist in the DB with a faster `Read` op to decide if they should Create (totally fresh) or Update (just TTL). These queries now hide this information and forces user to execute `Create` and on `AlreadyExists` execute `Update` (just TTL) instead.

**Comment by azdagron** [comment]
Hi @ryysud! Considering @amoore877's comments in #1017, we're thinking of no longer providing this feature. Before we do however, we wanted to check with you to see if this was solving some problem for you?

**Comment by azdagron** [comment]
@ryysud and I talked offline and agreed that we can close this PR. Even though we didn't end up taking it, really appreciate the contribution @ryysud!

**Comment by ryysud** [comment]
Thank you for your support @azdagron!

**Pull request closed without merging**